### ISSUE TITLE
[Feature] Automatic tuple name bindings

### ISSUE THREAD

**Issue by bradphelan** [dotnet/csharplang#591]
I just realized that this

            foreach (var o in from toolPathItem in controlPointSolvedInterpolator.GroupedKeyPoints.Concat()
                              from edgePosition in toolPathItem.ToolAxesSolved.RailPositionOption.AsEnumerable()
                              from edgeCurve in toolPathItem.ControlPoint.Run.EdgeCurveOption.AsEnumerable()
                              select (toolPathItem, edgePosition, edgeCurve))
                color = RenderPointOnEdge( renderMesh, o.toolPathItem, color, o.edgeCurve, o.edgePosition, d );

doesn't work but

            foreach (var o in from toolPathItem in controlPointSolvedInterpolator.GroupedKeyPoints.Concat()
                              from edgePosition in toolPathItem.ToolAxesSolved.RailPositionOption.AsEnumerable()
                              from edgeCurve in toolPathItem.ControlPoint.Run.EdgeCurveOption.AsEnumerable()
                              select new {toolPathItem, edgePosition, edgeCurve})
                color = RenderPointOnEdge( renderMesh, o.toolPathItem, color, o.edgeCurve, o.edgePosition, d );

does. I would have to do

            foreach (var o in from toolPathItem in controlPointSolvedInterpolator.GroupedKeyPoints.Concat()
                              from edgePosition in toolPathItem.ToolAxesSolved.RailPositionOption.AsEnumerable()
                              from edgeCurve in toolPathItem.ControlPoint.Run.EdgeCurveOption.AsEnumerable()
                              select (toolPathItem : toolPathItem, edgePosition:edgePosition, edgeCurve:edgeCurve))
                color = RenderPointOnEdge( renderMesh, o.toolPathItem, color, o.edgeCurve, o.edgePosition, d );

to get the tuple version working. This is a bit unfortunate as the ValueTuple version is more efficient but the code is much uglier. Can we please have automatic tuple name generation just like with anonymous types?

**Comment by jnm2** [comment]
Yes. Conversation here: https://github.com/dotnet/csharplang/issues/370
